v2/emailpassword/nextjs/app-directory/protecting-route.mdx

```diff
@@ -183,8 +184,21 @@ import { TryRefreshComponent } from "./tryRefreshClientComponent";
 // @ts-ignore
 import { SessionAuthForNextJS } from "./sessionAuthForNextJS";
 
+async function getSession(): Promise<{ session: SessionContainer | undefined; hasToken: boolean; hasInvalidClaims: boolean }> {
+    let session: SessionContainer | undefined;
+    let hasToken = false;
+    let hasInvalidClaims = false;
+    
+    try {
+        ({ session, hasToken, hasInvalidClaims } = await getSSRSession(cookies().getAll(), headers()));
+    } catch (error) {
+        // Handle the error here
+    }
+    return { session, hasToken, hasInvalidClaims };
+}
+
 export async function HomePage() {
-    const { session, hasToken, hasInvalidClaims } = await getSSRSession(cookies().getAll(), headers());
+    const { session, hasToken, hasInvalidClaims } = await getSession();
```

Comment:
- this type of code structure is not good, cause in the `// Handle the error here` part, the user can';t return a 500 error to the client since getSession does not return a next response.
- Also, the helper function named getSession is bad cause our sdk also has a function called getSession..

Instead, put all the logic in the `HomePage` component itself, and in the "// Handle the error here" part, send a 500 error to the client

Reply:
Updated the function name to `getSSRSessionHelper` and showing the error message in the UI if there is an error.